driver/src/main/scala/org/mongodb/scala/WriteConcern.scala

```diff
@@ -104,89 +107,11 @@ object WriteConcern {
-   */
-  def apply(w: Int, wtimeout: Int, fsync: Boolean): JWriteConcern = new JWriteConcern(w, wtimeout, fsync)
-
-  /**
-   * Creates a WriteConcern object.
-   *
-   * Specifies the number of servers to wait for on the write operation, and exception raising behavior
-   *
-   *  `w` represents the number of servers:
-   *
-   * - `w=-1` None, no checking is done
-   * - `w=0` None, network socket errors raised
-   * - `w=1` Checks server for errors as well as network socket errors raised
-   * - `w>1` Checks servers (w) for errors as well as network socket errors raised
+   * Constructs an instance with the given integer-based value for w and the given value for wTimeoutMS.
    *
-   *
-   * @param w        number of writes
-   * @param wtimeout timeout for write operation
-   * @param fsync    whether or not to fsync
-   * @param j        whether writes should wait for a journaling group commit
-   */
-  def apply(w: Int, wtimeout: Int, fsync: Boolean, j: Boolean): JWriteConcern = new JWriteConcern(w, wtimeout, fsync, j)
-
-  /**
-   * Creates a WriteConcern object, based on a tag name
-   *
-   * @param w        the tag name
-   * @param wtimeout timeout for write operation
-   * @param fsync    whether or not to fsync
-   * @param j        whether writes should wait for a journaling group commit
-   */
-  def apply(w: String, wtimeout: Int, fsync: Boolean, j: Boolean): JWriteConcern = new JWriteConcern(w, wtimeout, fsync, j)
-
-  /**
-   * Create a Majority Write Concern that requires a majority of servers to acknowledge the write.
-   *
-   * @param wtimeout timeout for write operation
-   * @param fsync    whether or not to fsync
-   * @param j        whether writes should wait for a journal group commit
-   * @return Majority, a subclass of WriteConcern that represents the write concern requiring most servers to acknowledge the write
-   */
-  def majorityWriteConcern(wtimeout: Int, fsync: Boolean, j: Boolean): JWriteConcern.Majority =
-    JWriteConcern.majorityWriteConcern(wtimeout, fsync, j)
-
-  /**
-   * A write concern that blocks acknowledgement of a write operation until a majority of replica set members have applied it.
+   * @param w          the w value, which must be >= 0
+   * @param wTimeout the wTimeout in milliseconds, which must be >= 0
    */
-  object Majority {
-    /**
-     * Create a new Majority WriteConcern.
-     */
-    def apply(): JWriteConcern.Majority = new JWriteConcern.Majority()
-
-    /**
-     * Create a new WriteConcern with the given configuration.
-     *
-     * @param wtimeout timeout for write operation
-     * @param fsync    whether or not to fsync
-     * @param j        whether writes should wait for a journaling group commit
-     */
-    def apply(wtimeout: Int, fsync: Boolean, j: Boolean): JWriteConcern.Majority = new JWriteConcern.Majority(wtimeout, fsync, j)
-  }
+  def apply(w: Int, wTimeout: Duration): JWriteConcern = new JWriteConcern(w, wTimeout.toMillis.toInt)
```

Comment:
Because those constructors don't exist in JWriteConcern, and all this did was wrap the original methods.

I might just add `withDuration` to the implicit and make it a chainable builder.